### DISCUSSION TITLE
Why group by operation return only 1 column?

### DISCUSSION THREAD

**Discussion by oleg-korshunov** [sqlalchemy/sqlalchemy#9724] [Usage Questions] [answered]
Can you please explain me why does grouping by operation return only 1 column if I specify two in the select? How can I return both columns?
```python
        number_of_boocked_room = (
            (
                select(Room.hotel_id.label("hotel_id"), func.count(Room.hotel_id).label("number_of_boocked_room"))
                .outerjoin(Booking, Booking.room_id == Room.id)
                .where(
                    ((date_from <= Booking.date_from) & (Booking.date_from <= date_to))
                    | (((date_from <= Booking.date_to) & (Booking.date_to <= date_to)))
                )
            ).group_by(Room.hotel_id)
            # .cte("number_of_boocked_room")
        )
        async with async_session_maker() as ssesion:
            result = await ssesion.scalars(number_of_boocked_room)
            result = result.all()
            print(result)
            return result
            
 # router
 @router.get("")
async def get_hotels_with_available_rooms(location: str, date_from: date = None, date_to: date = None):
    return await DaoHotel.get_hotels_with_available_rooms(location, date_from, date_to)
 ```
 ```sql
 SELECT room.hotel_id        AS hotel_id,
       Count(room.hotel_id) AS number_of_boocked_room
FROM   room
       LEFT OUTER JOIN booking
                    ON booking.room_id = room.id
WHERE  booking.date_from >= '2020-10-15'
       AND booking.date_from <= '2024-06-20'
        OR booking.date_to >= '2020-10-15'
           AND booking.date_to <= '2024-06-20'
GROUP  BY room.hotel_id 
```
![image](https://user-images.githubusercontent.com/44620818/235314804-76e059cf-dcc0-4eb6-b1ef-bf750ec65c0d.png)

![image](https://user-images.githubusercontent.com/44620818/235314701-260c411a-1f69-4365-b1be-cafa6e624757.png)

**Comment by CaselIT** [comment, marked as answer]
Hi,
Well you are calling `scalars()`. It behaviour is to return the first column only.

Use `execute()` instead

**Reply by oleg-korshunov** [reply]
But then I will have problems with the `paydantic`, what should I do in this case?

**Reply by CaselIT** [reply]
I'm not sure what you mean by "problems with pydantic". I don't see what issue would execute cause here.

Also you asked how to get two columns back, execute is the method to use if you want multiple columns back.

**Reply by oleg-korshunov** [reply]
Yes sure you right, thanks for the reply, I mean the `execution` return `ChunkedIteratorResult`, so in my case I suppose need rewrite query or figure out how to better work with it in `pydantic`

**Reply by CaselIT** [reply]
If you call `.all()` it will return a sequence of rows that are basically named tuples. With that you can create any other object you like

**Reply by oleg-korshunov** [reply]
Thank you for helping! I fixed my problem like this
```python
  async with async_session_maker() as ssesion:
      number_of_boocked_room = await ssesion.execute(number_of_boocked_room)
      result = [r._asdict() for r in number_of_boocked_room]
      return result
```
There may be a better solution, but I didn't find it

**Reply by CaselIT** [reply]
you may be able to use [`.mappings()`](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.mappings)

**Reply by oleg-korshunov** [reply]
Perfect! That's what I need
